app/src/main/java/com/movies/presentation/base/fragment/BaseChildFragment.kt

```diff
@@ -0,0 +1,69 @@
+package com.movies.presentation.base.fragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.movies.R
+import com.movies.common.extensions.replaceView
+import com.movies.databinding.BaseFragmentBinding
+import com.movies.presentation.home.ui.HomeFragment
+import com.movies.presentation.view.navigation.NavigationView
+
+class BaseChildFragment : Fragment() {
+
+    private lateinit var childBinding: BaseFragmentBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return BaseFragmentBinding.inflate(inflater, container, false).also {
+            childBinding = it
+        }.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setUpFragment()
+    }
+
+    private fun setUpFragment() {
+        childFragmentManager.beginTransaction().replace(
+            R.id.childFragmentContainerView, HomeFragment(),
+        ).commit()
+    }
+
+    fun onBackPress() {
+        if (currentFragment().needPressBack().not()) return
+
+        if (childFragmentManager.backStackEntryCount == 0) {
+            requireActivity().finish()
+            return
+        }
+        childFragmentManager.popBackStack()
+    }
+
+    private fun currentFragment() =
```

Comment:
change it to variable or rename function

Reply:
done

Reply:
@etozheluka @DavitTsiskarashvili setCurrentFragment is not the good name, because this functions behaviour is to get fragment not set, also whats the problem to be a function ?

Reply:
ფუნქცია რომ იყო არაა პრობლემა, უბრალოდ თვლიან რომ ფუნქციის სახელი ზმნით უნდა იწყებოდეს, რაც რეალურად არაა პრობლემა. determineCurrentFragment და მაგის ჯანი მაშინ

Reply:
არა რა determine